### PR TITLE
✨ (entity selector) feedback

### PR DESCRIPTION
**Pull request by sophiamersmann**
Small bug fixes and improvements coming out of design review:

(What Marwa said is in quotes)

- [x] "The hover effect of the close button is different from the designs"
- [x] "The hover effect is missing on the entity rows"
- [x] "The “close” button is distorted on mobile"
- [x] "It would be nice (but not a must) if after a user scrolls a lot, they can click on the are on top to scroll back to the top of the list"
- [x] "I feel like when we have radio buttons, the user needs some kind of feedback that they’ve selected an entity. So instead of closing the modal immediately after selection, it would be good to see it selected."
- [ ] "I think this was an issue in the past but the viewport does not zoom out after the user is done typing and selecting (i think it should)"
    - The only "fix" that I'm aware of is setting the font size to 16px which looks bad, so I guess we'll have to live with this one...
- [x] "“Sort by” should be Gray 70"

Not clear how to address:
- [ ] "The animation is really nice when the user is close to the “selected” list but once they get further, it gets a bot strange… I’m not 100% sure how that could be fixed but it does seem a bit choppy"
    - We can't really slow down the item that travels to the top because then it would take forever to get there... but I did try to slow down the items around the selected one, so that it's hopefully a bit more obvious that we just moved it to the top
- [x] "It might make sense to have the selection visible at the top of the list. I think it’s a bit weird that we have Germany and Europe that are not in alphabetical order (because of the current location) but then again the actual selection (which is the most important thing to see in this case) is buried somewhere else in the list."
    - I tried to make it so that, upon selecting, the previously selected item doesn't get animated and just disappears from the top and reappears in its original position, but that was even weirder than swapping..

----

To do before merging:
- [x] Replace `rem` with `px`
- ~Show "World" on top with a [globe icon](https://fontawesome.com/icons/globe?f=classic&s=solid)~ (was weird)

### CODE REVIEW

**Comment by sophiamersmann** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3501?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3501** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3501?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3500** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3500?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3490** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3490?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3466** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3466?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3443](https://github.com/owid/owid-grapher/pull/3443) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3443?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3452** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3452?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3447** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3447?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3436** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3436?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3435** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3435?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3434** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3434?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3373** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3373?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3353** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3353?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#3356](https://github.com/owid/owid-grapher/pull/3356) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3356?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3363](https://github.com/owid/owid-grapher/pull/3363) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3363?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sophiamersmann and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx [hunk 1]
```diff
@@ -591,7 +602,11 @@ export class EntitySelector extends React.Component<{
             this.selectionArray.toggleSelection(entityName)
         } else {
             this.selectionArray.setSelectedEntities([entityName])
-            if (this.props.onDismiss) this.props.onDismiss()
+            if (this.props.onDismiss) {
+                this.dismissTimer = setTimeout(() => {
+                    if (this.props.onDismiss) this.props.onDismiss()
+                }, 250)
```
Comment (marcelgerber) [inline packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx:608]: I see Marwa's comment that was suggesting this, but I think this now feels even weirder than it did before.
You can just about glimpse that something is animating, but then it the modal just closes on you without notice, and that feels quite broken.

File: packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx [hunk 2]
```diff
@@ -947,48 +981,35 @@ export class EntitySelector extends React.Component<{
...
-                            <>
-                                Current selection:
-                                <span className="entity-name">
-                                    {selectedEntityNames[0]}
-                                </span>
-                            </>
-                        ) : (
-                            "Empty selection"
-                        )}
-                    </div>
```
Comment (marcelgerber) [inline packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx:982]: Was removing this also due to Marwa's feedback? I felt like it worked quite well.

**Review by marcelgerber — APPROVED** [review]

**Comment by sophiamersmann** [comment]
### Merge activity

* **May 3, 4:04 AM EDT**: @sophiamersmann started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3501).

**Pull request merged**